src/Microsoft.AspNet.Mvc.TagHelpers/EnvironmentTagHelper.cs

```diff
@@ -67,25 +67,33 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 return;
             }
 
-            var environments = Names.Split(NameSeparator, StringSplitOptions.RemoveEmptyEntries)
-                                    .Where(name => !string.IsNullOrWhiteSpace(name));
-
-            if (!environments.Any())
+            var currentEnvironmentName = HostingEnvironment.EnvironmentName?.Trim();
+            if (string.IsNullOrWhiteSpace(currentEnvironmentName))
             {
-                // Names contains only commas or empty entries, do nothing
+                // No current environment name, do nothing
                 return;
             }
 
-            var currentEnvironmentName = HostingEnvironment.EnvironmentName?.Trim();
+            var values = Names.Split(NameSeparator, StringSplitOptions.RemoveEmptyEntries);
+            var environments = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why worry much about extra runtime checks if the user is silly enough to repeat environment names in the attribute?